src/lib/operators/table_scan/column_vs_value_table_scan_impl.cpp

```diff
@@ -127,6 +133,51 @@ void ColumnVsValueTableScanImpl::_scan_dictionary_segment(const BaseDictionarySe
   });
 }
 
+void ColumnVsValueTableScanImpl::_scan_sorted_segment(const BaseSegment& segment, const ChunkID chunk_id,
+                                                      PosList& matches,
+                                                      const std::shared_ptr<const PosList>& position_filter,
+                                                      const OrderByMode ordered_by) const {
+  resolve_data_and_segment_type(segment, [&](const auto type, const auto& typed_segment) {
+    using ColumnDataType = typename decltype(type)::type;
+
+    if constexpr (std::is_same_v<decltype(typed_segment), const ReferenceSegment&>) {
```

Comment:
I prefer `std::is_same_v<std::decay_t<decltype(typed_segment)>, ReferenceSegment>` because it expresses that we don't care about the constness and references (which might change).